ccl/storageccl: TestRandomKeyAndTimestampExport failed

[(ccl/storageccl).TestRandomKeyAndTimestampExport failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1733479&tab=buildLog) on [master@e2dcdf142961200ec8706154be08d1b1f09ce851](https://github.com/cockroachdb/cockroach/commits/e2dcdf142961200ec8706154be08d1b1f09ce851):

```
=== RUN   TestRandomKeyAndTimestampExport
I200206 16:42:09.936138 1777 storage/engine/rocksdb.go:603  opening rocksdb instance at "/tmp/TestRandomKeyAndTimestampExport612683290"
--- FAIL: TestRandomKeyAndTimestampExport (231.14s)
=== RUN   TestRandomKeyAndTimestampExport/targetSize=65536/ts_(0-∞],_all,_nontimebound
        --- FAIL: TestRandomKeyAndTimestampExport/targetSize=65536/ts_(0-∞],_all,_nontimebound (1.58s)
            require.go:1104: 
                	Error Trace:	export_test.go:380
                	Error:      	Expect "<nil>" to match "export size \(65536 bytes\) exceeds max size \(65536 bytes\)"
                	Test:       	TestRandomKeyAndTimestampExport/targetSize=65536/ts_(0-∞],_all,_nontimebound
=== RUN   TestRandomKeyAndTimestampExport/targetSize=65536/ts_(0-∞],_all,_timebound
        --- FAIL: TestRandomKeyAndTimestampExport/targetSize=65536/ts_(0-∞],_all,_timebound (1.29s)
            require.go:1104: 
                	Error Trace:	export_test.go:380
                	Error:      	Expect "<nil>" to match "export size \(65536 bytes\) exceeds max size \(65536 bytes\)"
                	Test:       	TestRandomKeyAndTimestampExport/targetSize=65536/ts_(0-∞],_all,_timebound
=== RUN   TestRandomKeyAndTimestampExport/targetSize=65536
    --- FAIL: TestRandomKeyAndTimestampExport/targetSize=65536 (6.17s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRandomKeyAndTimestampExport PKG=./pkg/ccl/storageccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRandomKeyAndTimestampExport.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/storageccl).TestRandomKeyAndTimestampExport failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1742619&tab=buildLog) on [master@1551d49cc9abcc791bd7295b01d5a16a5fd93292](https://github.com/cockroachdb/cockroach/commits/1551d49cc9abcc791bd7295b01d5a16a5fd93292):

```
=== RUN   TestRandomKeyAndTimestampExport
I200210 23:47:05.972733 2880 storage/engine/rocksdb.go:603  opening rocksdb instance at "/tmp/TestRandomKeyAndTimestampExport780518863"
--- FAIL: TestRandomKeyAndTimestampExport (239.82s)
=== RUN   TestRandomKeyAndTimestampExport/targetSize=65536/ts_(0-∞],_all,_nontimebound
        --- FAIL: TestRandomKeyAndTimestampExport/targetSize=65536/ts_(0-∞],_all,_nontimebound (1.13s)
            require.go:1104: 
                	Error Trace:	export_test.go:461
                	Error:      	Expect "<nil>" to match "export size \(65536 bytes\) exceeds max size \(65536 bytes\)"
                	Test:       	TestRandomKeyAndTimestampExport/targetSize=65536/ts_(0-∞],_all,_nontimebound
=== RUN   TestRandomKeyAndTimestampExport/targetSize=65536/ts_(0-∞],_all,_timebound
        --- FAIL: TestRandomKeyAndTimestampExport/targetSize=65536/ts_(0-∞],_all,_timebound (1.02s)
            require.go:1104: 
                	Error Trace:	export_test.go:461
                	Error:      	Expect "<nil>" to match "export size \(65536 bytes\) exceeds max size \(65536 bytes\)"
                	Test:       	TestRandomKeyAndTimestampExport/targetSize=65536/ts_(0-∞],_all,_timebound
=== RUN   TestRandomKeyAndTimestampExport/targetSize=65536
    --- FAIL: TestRandomKeyAndTimestampExport/targetSize=65536 (4.87s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRandomKeyAndTimestampExport PKG=./pkg/ccl/storageccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRandomKeyAndTimestampExport.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/storageccl).TestRandomKeyAndTimestampExport failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1742760&tab=buildLog) on [master@d83713fb7d9d52e45e488bf5c56e22cf1041fdc6](https://github.com/cockroachdb/cockroach/commits/d83713fb7d9d52e45e488bf5c56e22cf1041fdc6):

```
=== RUN   TestRandomKeyAndTimestampExport
I200211 04:04:24.604099 3000 storage/engine/rocksdb.go:603  opening rocksdb instance at "/tmp/TestRandomKeyAndTimestampExport994232286"
--- FAIL: TestRandomKeyAndTimestampExport (295.10s)
=== RUN   TestRandomKeyAndTimestampExport/targetSize=1048576/ts_(0-∞],_all,_nontimebound
        --- FAIL: TestRandomKeyAndTimestampExport/targetSize=1048576/ts_(0-∞],_all,_nontimebound (0.33s)
            require.go:1104: 
                	Error Trace:	export_test.go:461
                	Error:      	Expect "<nil>" to match "export size \(1048576 bytes\) exceeds max size \(1048576 bytes\)"
                	Test:       	TestRandomKeyAndTimestampExport/targetSize=1048576/ts_(0-∞],_all,_nontimebound
=== RUN   TestRandomKeyAndTimestampExport/targetSize=1048576/ts_(0-∞],_all,_timebound
        --- FAIL: TestRandomKeyAndTimestampExport/targetSize=1048576/ts_(0-∞],_all,_timebound (0.34s)
            require.go:1104: 
                	Error Trace:	export_test.go:461
                	Error:      	Expect "<nil>" to match "export size \(1048576 bytes\) exceeds max size \(1048576 bytes\)"
                	Test:       	TestRandomKeyAndTimestampExport/targetSize=1048576/ts_(0-∞],_all,_timebound
=== RUN   TestRandomKeyAndTimestampExport/targetSize=1048576
    --- FAIL: TestRandomKeyAndTimestampExport/targetSize=1048576 (3.09s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRandomKeyAndTimestampExport PKG=./pkg/ccl/storageccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRandomKeyAndTimestampExport.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/ccl/storageccl/export_test.go
+++ b/pkg/ccl/storageccl/export_test.go
@@ -446,15 +446,15 @@ func assertEqualKVs(
 					lessThisKey := dataSizeWhenExceeded - uint64(len(kv.Key.Key)+len(kv.Value))
 					if lessThisKey >= maxSize {
 						dataSizeWhenExceeded = lessThisKey
-						// It might be the case that this key would lead to an SST of exactly
-						// max size, in this case we overwrite max size to be less so that
-						// we still generate an error.
-						if dataSizeWhenExceeded == maxSize {
-							maxSize--
-						}
 					} else {
 						break
 					}
+				}
+				// It might be the case that this key would lead to an SST of exactly
+				// max size, in this case we overwrite max size to be less so that
+				// we still generate an error.
+				if dataSizeWhenExceeded == maxSize {
+					maxSize--
 				}
 				_, _, _, err = e.ExportToSst(prevStart, endKey, startTime, endTime,
 					exportAllRevisions, targetSize, maxSize, io)

